### ISSUE TITLE
System includes should be excluded from clang-tidy

### ISSUE THREAD

**Issue by RudolfWeeber** [espressomd/espresso#3526]
So it can be used in the standard development workflow, not involvin docker or the like

**Comment by RudolfWeeber** [comment]
system includes are actually excluded, but boost is not.

On the command-line this can be achieved with
-line-filter="[{name: boost/mpi/collectives/all_reduce.hpp}]"
I'd much prefer this to patching boost in CI docker containers.

BTW: Are we sure, .clang-tidy is found and being used for out-of-tree builds?

**Comment by KaiSzuttor** [comment]
> BTW: Are we sure, .clang-tidy is found and being used for out-of-tree builds?

yes, see below

[clang homepage](https://clang.llvm.org/extra/clang-tidy/#suppressing-undesired-diagnostics)
```
 --config=<string>              -
                                   Specifies a configuration in YAML/JSON format:
                                     -config="{Checks: '*',
                                               CheckOptions: [{key: x,
                                                               value: y}]}"
                                   When the value is empty, clang-tidy will
                                   attempt to find a file named .clang-tidy for
                                   each source file in its parent directories.
```

**Comment by jngrad** [comment]
This issue was resolved by #3684.
